include/bluetooth/hci.h

```diff
@@ -1825,6 +1825,49 @@ typedef bool bt_hci_vnd_evt_cb_t(struct net_buf_simple *buf);
   */
 int bt_hci_register_vnd_evt_cb(bt_hci_vnd_evt_cb_t cb);
 
+
+/** Allocate a HCI event buffer.
+ *
+ * This function allocates a new buffer for a HCI event. It is given the
+ * OpCode (encoded e.g. using the BT_OP macro) and the total length of the
+ * parameters. Upon successful return the buffer is ready to have
+ * the parameters encoded into it.
+ *
+ * @param evt        Event OpCode.
+ * @param len        Length of event parameters.
+ *
+ * @return Newly allocated buffer.
+ */
+struct net_buf *bt_hci_evt_create(u8_t evt, u8_t len);
+
+/** Allocate a HCI Command Complete event buffer.
+ *
+ * This function allocates a new buffer for HCI Command Complete event.
+ * It is given the OpCode (encoded e.g. using the BT_OP macro) and the total
+ * length of the parameters. Upon successful return the buffer is ready to have
+ * the parameters encoded into it.
+ *
+ * @param op         Command OpCode.
+ * @param plen       Length of command parameters.
+ *
+ * @return Newly allocated buffer.
+ */
+struct net_buf *bt_hci_cmd_complete(u16_t op, u8_t plen);
```

Comment:
Please be consistent with the choice of verb since you're making these public APIs. You have "create" with `bt_hci_evt_create()` but this one is missing a verb completely. Same goes for `bt_hci_cmd_status()`.

I'm a little bit doubtful whether we need these at all TBH since they're so simple (i.e. users could just use the bt_buf APIs directly instead).

Reply:
They would be duplicated though, does that bt_hci_cmd_complete_create makes it sounds any better?

Reply:
I think `bt_hci_create_cmd_complete` is the naming convention, but that would mean `bt_hci_evt_create` is not following the same naming convention.

Reply:
No bt_hci_cmd_create, bt_hci_cmd_snd, etc, the verb always comes last so the convention is the other way around at least when it comes to hci_raw.